msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/atpop/TestCase1922511.java

```diff
@@ -0,0 +1,155 @@
+import com.microsoft.identity.client.msal.automationapp.sdk.MsalSdk;
+import com.microsoft.identity.client.msal.automationapp.testpass.broker.AbstractMsalBrokerTest;
+import com.microsoft.identity.client.ui.automation.TokenRequestTimeout;
+import com.microsoft.identity.client.ui.automation.constants.AuthScheme;
+import com.microsoft.identity.client.ui.automation.interaction.OnInteractionRequired;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.common.java.exception.ServiceException;
+import com.microsoft.identity.common.java.providers.oauth2.IDToken;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+
+// [Non-Joined] Acquire PoP token interactive followed by Silent
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1922511
+public class TestCase1922511 extends AbstractMsalBrokerTest {
+    @Test
+    public void test_1922511() throws Throwable {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.LOGIN)
+                .authScheme(AuthScheme.POP)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                        .prompt(PromptParameter.LOGIN)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .speedBumpExpected(false)
+                        .broker(mBroker)
+                        .expectingBrokerAccountChooserActivity(false)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+        verifyATForPop(authResult);
```

Comment:
Could we place this method in one of the super classes so it doesn't have to be defined in each case?

Reply:
+1 to that. I'm actually thinking that we may want to put this directly on the `MsalAuthResult` class and then we can just call it like `authResult.verifyAtForPop()`

Reply:
done, added this method to MsalAuthResult. I did not use asserts here as we have to add the dependency in this pkg for this purpose. Hence verified the negation of each expectation and returned the final result as string. If the string is returned as "success" -> it means that the AT-POP was successfully verified.

Reply:
I'm not sure I follow...did you push the commit already? (can't see it in GitHub)

Reply:
In regards to the dependency, I think you can put the verifyAtPop method directly in the parent class i.e. AuthResult and that seems to already have the assertion dependency

Reply:
I think even if i do this, it would expect a dependency to be added to use jwtParser. I wanted to avoid adding it